pipeline/load.py

```diff
@@ -0,0 +1,291 @@
+    """
+    Inserts genres into the database they don't already exist.
+    Logs whether the genre was added or already exists.
+    Returns the genre id if inserted/exists.
+    Returns -1 in the case of an error.
+    """
+    try:
+        check_query = "SELECT genre_id FROM genre WHERE genre_name = %s LIMIT 1;"
+        cursor.execute(check_query, (genre_name,))
+        genre_id = cursor.fetchone()
+
+        if genre_id:
+            logging.info(
+                "Genre '%s' already exists in the database.", genre_name)
+            return genre_id[0]
+
+        insert_query = """INSERT INTO genre (genre_name) VALUES (%s)
+                            RETURNING genre_id;"""
+        cursor.execute(insert_query, (genre_name,))
+        logging.info("Genre '%s' was added to the database.", genre_name)
+        genre_id = cursor.fetchone()
+        return genre_id[0]
+
+    except psycopg2.Error:
+        logging.error("Error inserting genre: '%s'", genre_name)
+        return -1
+
+
+def insert_release(release_name: str, release_date: datetime,
+                   release_type: str, artist_name: str,
+                   cursor: extensions.cursor) -> int:
+    """
+    Inserts a release into the database if it doesn't already exist.
+    Logs whether the release was added or already exists.
+    Returns the release id if inserted/exists.
+    Returns -1 in the case of an error.
+    """
+    artist_id = get_id_from_artist(artist_name, cursor)
+    type_id = get_id_from_release_type(release_type, cursor)
+
+    if artist_id == -1:
+        logging.info("Error with provided artist name: %s", artist_name)
+        return -1
+
+    if type_id == -1:
+        logging.info("Error with provided release_type: %s", release_type)
+        return -1
+
+    try:
+        check_query = """SELECT release_id FROM release
+                        WHERE (release_name = %s
+                        AND release_date = %s
+                        AND artist_id = %s
+                        AND type_id = %s);"""
+
+        cursor.execute(check_query, (release_name,
+                       release_date, artist_id, type_id))
+        release_id = cursor.fetchone()
+
+        if release_id:
+            logging.info(
+                "'%s' already exists with release ID %s.", release_name, release_id[0])
+            return release_id[0]
+
+        insert_query = """INSERT INTO release (release_name, release_date, artist_id, type_id)
+                            VALUES (%s, %s, %s, %s)
+                            RETURNING release_id;"""
+        cursor.execute(insert_query, (release_name,
+                                      release_date, artist_id, type_id))
+        release_id = cursor.fetchone()
+        logging.info(
+            "Release '%s' of '%s' was added to the database.", release_id, release_name)
+        return release_id[0]
+
+    except psycopg2.Error:
+        logging.error("Error inserting release: %s", release_name)
+        return -1
```

Comment:
This function is very large, could we maybe split it into check_exists() and insert_release()

Reply:
You have resolved a few comments but not changed anything - if you disagree we can discuss on a huddle